tests/NewZealand/AnzacDayTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ *  This file is part of the Yasumi package.
+ *
+ *  Copyright (c) 2015 - 2016 AzuyaLabs
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ *  @author Sacha Telgenhof <stelgenhof@gmail.com>
+ */
+
+namespace Yasumi\Tests\NewZealand;
+
+use DateTime;
+use DateTimeZone;
+
+/**
+ * Class for testing ANZAC day in the New Zealand.
+ */
+class AnzacDayTest extends NewZealandBaseTestCase
+{
+    /**
+     * The name of the holiday
+     */
+    const HOLIDAY = 'anzacDay';
+
+    /**
+     * Tests ANZAC Day
+     *
+     * @dataProvider HolidayDataProvider
+     *
+     * @param int $year the year for which the holiday defined in this test needs to be tested
+     * @param DateTime $expected the expected date
+     */
+    public function testHoliday($year, $expected)
+    {
+        $this->assertHoliday(self::REGION, self::HOLIDAY, $year,
+            new DateTime($expected, new DateTimeZone(self::TIMEZONE)));
+    }
+
+    /**
+     * Returns a list of test dates
+     *
+     * @return array list of test dates for the holiday defined in this test
+     */
+    public function HolidayDataProvider()
+    {
+        return [
```

Comment:
@stelgenhof These lists are actually generated, I didn't write them by hand don't worry :)
My reasoning for hard coding them was that the code used to generate them seemed a bit too complex for unit tests, it resembled the calculation methods in the provider; to me it made sense to keep the tests as simple as possible.

But that's no problem I can use the code I generated the lists with instead.